src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -0,0 +1,614 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A synthesized type used for collection expressions where the target type
+    /// is IEnumerable&lt;T&gt;, IReadOnlyCollection&lt;T&gt;, or IReadOnlyList&lt;T&gt;.
+    /// If the collection expression has a known length, the type is generated with
+    /// an array field; otherwise the type is generated with a List&lt;T&gt; field.
+    /// <code>
+    /// sealed class &lt;&gt;z__ReadOnlyArray&lt;T&gt; { private readonly T[] _items; }
+    /// sealed class &lt;&gt;z__ReadOnlyList&lt;T&gt; { private readonly List&lt;T&gt; _items; }
+    /// </code>
+    /// </summary>
+    internal sealed class SynthesizedReadOnlyListTypeSymbol : NamedTypeSymbol
+    {
+        private static readonly SpecialType[] s_requiredSpecialTypes = new[]
+        {
+            SpecialType.System_Collections_IEnumerable,
+            SpecialType.System_Collections_Generic_IEnumerable_T,
+            SpecialType.System_Collections_Generic_IReadOnlyCollection_T,
+            SpecialType.System_Collections_Generic_IReadOnlyList_T,
+            SpecialType.System_Collections_Generic_ICollection_T,
+            SpecialType.System_Collections_Generic_IList_T,
+        };
+
+        private static readonly WellKnownType[] s_requiredWellKnownTypes = new[]
+        {
+            WellKnownType.System_Collections_Generic_List_T,
+        };
```

Comment:
:question: I think this is unintentionally left unused.

Reply:
Yes, thanks, this item was unnecessary (and should have been removed) since line 118 checks for this well-known type directly. The item has since been removed in [https://github.com/dotnet/roslyn/pull/70288](https://github.com/dotnet/roslyn/pull/70288).